### PR TITLE
color_files() never returns None for token; 

### LINKED ISSUE #3474: "Unhandled exception" when "cd ~"
With an empty configuration file, when doing a simple `cd ~`, the following backtrace is displayed in the terminal:
```
Unhandled exception in event loop:
  File "/usr/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/eventloop/utils.py", line 72, in schedule
    func()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 456, in redraw
    self._redraw()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 523, in _redraw
    self.context.run(run_in_context)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 509, in run_in_context
    self.renderer.render(self, self.layout)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/renderer.py", line 568, in render
    layout.container.preferred_height(size.columns, size.rows).preferred,
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 324, in preferred_height
    dimensions = [
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 325, in <listcomp>
    c.preferred_height(width, max_available_height) for c in self._all_children
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 792, in preferred_height
    return self.content.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 324, in preferred_height
    dimensions = [
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 325, in <listcomp>
    c.preferred_height(width, max_available_height) for c in self._all_children
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 2595, in preferred_height
    return self.content.preferred_height(width, max_available_height)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1620, in preferred_height
    return self._merge_dimensions(
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1648, in _merge_dimensions
    preferred = get_preferred()
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/containers.py", line 1613, in preferred_content_height
    return self.content.preferred_height(
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 639, in preferred_height
    content = self.create_content(width, height=1)  # Pass a dummy '1' as height.
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 787, in create_content
    cursor_position=translate_rowcol(
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 770, in translate_rowcol
    return Point(x=get_processed_line(row).source_to_display(col), y=row)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/layout/controls.py", line 716, in get_processed_line
    processed_line = transform(i, get_line(i))
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 319, in get_line
    for num, line in generator:
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 270, in create_line_generator
    yield from enumerate(split_lines(list(get_text_fragments())), start_lineno)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 268, in get_text_fragments
    yield _token_cache[t], v
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/lexers/pygments.py", line 141, in __missing__
    result = "class:" + pygments_token_to_classname(key)
  File "/usr/lib/python3.8/site-packages/prompt_toolkit/styles/pygments.py", line 66, in pygments_token_to_classname
    parts = ("pygments",) + token

Exception can only concatenate tuple (not "NoneType") to tuple
Press ENTER to continue...
```
Python: 3.8.2
xonsh: 0.9.14
Arch linux

### PR DESCRIPTION
**Pull request by bobhy**
Fix #3474 

Need code review in tests/test_pyghooks.py. Newly added test cases failing apparently because events not firing.  However, the events seem to be firing correctly when Xonsh runs interactively.

### CODE REVIEW

**Comment by bobhy** [comment]
Test test_color_on_lscolors_change in tests/test_ptk_highlight.py is failing because events.on_lscolors_change not firing in test environment.  It seems to be firing in interactive use.  Any suggestions to get events to work in pytest?

**Comment by scopatz** [comment]
Thanks for putting this in @bobhy! The test failure on CI seems correct

**Comment by necaris** [comment]
I was about to open a similar PR! With this fix in place, file colorization / tab-completion works as expected, but terminal apps like `vim` and `emacs -nw` stop working. Running `vi` shows a flash of content and then clears.

**Comment by bobhy** [comment]
@necaris, can you get more detail on what's happening with these apps? I use vim on a daily basis with this PR applied, "it works on my machine".

**Comment by bobhy** [comment]
@scopatz, I agree the assert fail is correct, but can't figure out whether I have initialized the test environment correctly for the UIT.   Am I initializing ptk2 shell and pygments correctly? I can't find another test case to copy.  I did try copying suggestive snippets from events test cases, but realized that would not be enough environment for my UIT.  So I'm stuck with code that works interactively and appears to comport to the API, but won't pass tests.  Suggestions from wiser heads eagerly solicited!

File: tests/test_ptk_highlight.py [hunk 1]
```diff
@@ -142,16 +142,22 @@ def test_nested():
         ],
     )
 
-
-@skip_if_on_windows
-def test_path(tmpdir, xonsh_builtins):
-
+@pytest.fixture
+def xonsh_builtins_LS_COLORS(xonsh_builtins):
```
Comment (scopatz) [inline tests/test_ptk_highlight.py]: please use only lowercase in function names

**Comment by scopatz** [comment]
It actually just seems like the comparison is not correct.  You seem to be comparing a `(token, str)` tuple to a list of token tuples.

**Comment by scopatz** [comment]
It seems like you are comparing an element of a list to the whole list

**Comment by necaris** [comment]
@bobhy I'm afraid I don't know where I'd start figuring it out :cry: I've uploaded a brief screencast of what happens when I try to run `vim`: https://drive.google.com/open?id=1N0u8KrPqmkiQEuI0LTxCJ6qzdKJLlBRW

**Comment by bobhy** [comment]
@scopatz, thanks for looking, but I believe the expression is correct.  I tested locally by changing the (Color.Green, ...) to (Color.BOLD_BLUE, ...) and now the test case passes (false positive).  (Plus I borrowed this code from `test_path()` immediately above, where it works as well.)  No, the problem in `test_color_on_lscolors_change' is that the event handler `on_lscolors_change` in `xonsh.pyghooks` is not getting registered, so the changes to LS_COLORS items don't get processed.  I've verified the event is fired in `xonsh.environ`, but it doesn't find any event handler to dispatch.  I see that test fixture dummy_shell has code that thinks it's processing real events, but I'm clearly missing some nuance of the test environment...

**Comment by bobhy** [comment]
Ended up having to have the test case define the on_lscolors_change event handler directly, so now can directly test for missing file type in LS_COLORS, and that test passes.  I believe this PR is ready for merge.

Still not sure why adding an `import xonsh.pyghooks` at the top of the test file didn't define the event handler like the interactive environment does.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**